### DISCUSSION TITLE
Unable to use static path in api route

### DISCUSSION THREAD

**Discussion by baymac** [vercel/next.js#27291] [Help]
I want to send verification email using an api route, I am unable to access the email template html in the api route. 

Email template inside `/public/emailTemplates` -> https://github.com/baymac/baymac.io/tree/main/public/emailTemplates

Loading email template here -> https://github.com/baymac/baymac.io/blob/main/lib/subscription/sendVerificationMail.ts#L26

I have tried paths:

`/vercel/path0/emailTemplates/emailconfirmation.html`
`/var/task/emailTemplates/emailconfirmation.html`
`/emailTemplates/emailconfirmation.html`
`../../public/emailTemplates/emailconfirmation.html`
`../../emailTemplates/emailconfirmation.html`
but none of them work.

Here is the api route -> https://github.com/baymac/baymac.io/blob/main/pages/api/newsletter/subscribe.ts

I think path resolution for static files in server side is different, can someone help me find it?

**Comment by wwwdepot** [comment]
just use `public/path/to/file`

```
  const body = fs
    .readFileSync(path.resolve("public/assets/email-template.html"))
    .toString();
```

**Reply by baymac** [reply]
Thanks for the suggestion, tried this (https://github.com/baymac/baymac.io/commit/d9796160be7cc2e6ca8c424495ca8ef59c833f49) but getting same response:
```
Some error occurred: ENOENT: no such file or directory, open '/var/task/public/emailTemplates/emailconfirmation.html'
```

Although this works in local.

**Reply by wwwdepot** [reply]
See if [this](https://github.com/vercel/next.js/issues/8251#issuecomment-854148718) helps

**Reply by baymac** [reply]
Thanks that fixed the issue, minor changes to the config would be required if you are using Next.js 11+.

```diff
const CopyPlugin = require("copy-webpack-plugin")

module.exports = {
+   target: "server",
+   webpack5: true,
-   target: "serverless",
-   future: {
-      webpack5: true,
-   }
    webpack: function (config, { dev, isServer }) {
        // Fixes npm packages that depend on `fs` module
        if (!isServer) {
            config.resolve.fallback.fs = false
        }
        // copy files you're interested in
        if (!dev) {
            config.plugins.push(
                new CopyPlugin({
                    patterns: [{ from: "public/emailTemplates", to: "emailTemplates/" }],
                })
            )
        }

        return config
    },
}
```

**Comment by ijjk** [comment]
Hi, related documentation for reading files can be found here https://nextjs.org/docs/basic-features/data-fetching#reading-files-use-processcwd do note there is a related issue with reading local files with webpack 5 here https://github.com/vercel/next.js/issues/24700
